include/robot_interfaces/monitored_robot_driver.hpp

```diff
@@ -168,9 +168,17 @@ class MonitoredRobotDriver : public RobotDriver<Action, Observation>
         // loop until shutdown and monitor action timing
         for (size_t t = 0; !is_shutdown_; t++)
         {
-            bool action_has_ended_on_time =
-                action_end_logger_.wait_for_timeindex(t,
-                                                      max_action_duration_s_);
+	  bool action_has_ended_on_time;
+	  if (max_action_duration_s_ >0)
+	    {
+	      action_has_ended_on_time = action_end_logger_.wait_for_timeindex(t,
+									       max_action_duration_s_);
+	    }
+	  else
+	    {
+	      action_has_ended_on_time = action_end_logger_.wait_for_timeindex(t);
+	    }
```

Comment:
i would not handle this as a special case, and setting it to zero for allowing arbitrary action duration does not seem very logical to me. the idea is that you set it to infinity if you do not want it to be checked, then no special case needs to be added.

Reply:
@mwuethri setting the durations to infinity seems indeed cleaner.